legal-api/tests/unit/resources/v2/test_business_filings/test_filings.py

```diff
@@ -1429,61 +1430,82 @@ def test_rules_memorandum_in_sr(session, mocker, requests_mock, client, jwt, ):
     assert rv.json.get('errors')[0].get('error') == error
 
 
-def test_resubmit_filing(session, client, jwt, mocker):
-    """Assert that the a filing can be resubmitted."""
-    # filing with awaiting review
+@pytest.mark.parametrize(
+    'filing_status, review_status',
+    [
+        (Filing.Status.DRAFT.value, None),
+        (Filing.Status.CHANGE_REQUESTED.value, ReviewStatus.CHANGE_REQUESTED),
+        (Filing.Status.APPROVED.value, ReviewStatus.APPROVED),
+    ]
+)
+def test_submit_or_resubmit_filing(session, client, jwt, mocker, requests_mock, filing_status, review_status):
+    """Assert that the a filing can be submitted/resubmitted."""
     identifier = 'Tb31yQIuBw'
     temp_reg = RegistrationBootstrap()
     temp_reg._identifier = identifier
     temp_reg.save()
     json_data = copy.deepcopy(CONTINUATION_IN_FILING_TEMPLATE)
-    filing = factory_pending_filing(None, json_data)
+    filing = factory_filing(None, json_data)
     filing.temp_reg = identifier
-    filing.payment_completion_date = datetime.now(timezone.utc)
-    filing._status = Filing.Status.AWAITING_REVIEW.value
+    filing._status = filing_status
     filing.save()
 
-    review = Review()
-    review.filing_id = filing.id
-    review.nr_number = json_data['filing']['continuationIn']['nameRequest']['nrNumber']
-    review.identifier = json_data['filing']['continuationIn']['foreignJurisdiction']['identifier']
-    review.completing_party = 'completing party'
-    review.status = ReviewStatus.AWAITING_REVIEW
-    review.save()
-
-    # filing with change requested
-    staff = User(username='staff_username',
-                 firstname='staff firstname',
-                 middlename='staff middlename',
-                 lastname='staff lastname',
-                 sub='sub',
-                 iss='iss',
-                 idp_userid='123',
-                 login_source='IDIR')
-    staff.save()
-
-    change_requested = ReviewResult()
-    change_requested.review_id = review.id
-    change_requested.comments = 'do the change'
-    change_requested.status = ReviewStatus.CHANGE_REQUESTED
-    change_requested.reviewer_id = staff.id
-    change_requested.submission_date = review.submission_date
-    review.review_results.append(change_requested)
-    review.status = ReviewStatus.CHANGE_REQUESTED
-    review.save()
-
-    filing.set_review_decision(Filing.Status.CHANGE_REQUESTED.value)
+    if filing_status != Filing.Status.DRAFT.value:
```

Comment:
I always prefer checking the positive conditions, since a negative condition could cause errors if, say, we added a filing status. Up to you whether to change or leave this.